lib/db.js

```diff
@@ -0,0 +1,173 @@
+import knex from 'knex';
+import log from './logger';
+import { DataStream } from 'scramjet';
+import { promisify } from 'util';
+import moment from 'moment';
+
+let _pg = null;
+function getDB () {
+  let knexConfig;
+  _pg = _pg || (async () => {
+    try {
+      knexConfig = module.parent.require('../knexfile-local');
+      log.info('Using connection data from knexfile-local');
+    } catch (e) {
+      knexConfig = module.parent.require('../knexfile');
+      log.info('Using connection data from production knexfile');
+    }
+
+    const pg = knex(knexConfig);
+    log.info('Connecting to database.');
+
+    return pg;
+  })()
+    .then(async pg => {
+      await pg.migrate.latest(knexConfig);
+      log.info('Database connected and migrations are handled, ready to roll!');
+      return pg;
+    });
+  // Run any needed migrations and away we go
+  return _pg;
+}
+
+/**
+ * Build an object that can be inserted into our database.
+ * @param {object} m measurement object
+ * @return {object} an object capable of being saved into the PostgreSQL database
+ */
+function convertMeasurementToSQLObject (stream, pg) {
```

Comment:
Hi @MichalCz, looking over this this morning, and saw the following behavior when trying to run using the local database.

```
$ node index.js -s 'StateAir_Hanoi'                                                               [±:signicode-feature/streamed-libs:∂]
2018-09-24T11:58:49.974Z - info: --- Full fetch started. ---
2018-09-24T11:58:49.988Z - info: Looking up source StateAir_Hanoi
2018-09-24T11:58:50.547Z - info: Getting stream for "StateAir_Hanoi" from "stateair"
2018-09-24T11:58:51.018Z - info: Using connection data from knexfile.local
2018-09-24T11:58:51.112Z - info: Connecting to database.
2018-09-24T11:58:51.338Z - info: Database connected and migrations are handled, ready to roll!
2018-09-24T11:58:51.374Z - error:  message=insert into "measurements" ("attribution", "averagingPeriod", "city", "coordinates", "country", "date", "location", "mobile", "parameter", "sourceName", "sourceType", "unit", "value") values ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13) returning "location" - column "attribution" of relation "measurements" does not exist, stack=error: column "attribution" of relation "measurements" does not exist
```

This is odd because https://github.com/openaq/openaq-fetch/pull/520/files#diff-be228df81b99a384aa50ad50bde7fca0R38 looks to be doing what I expect it should be doing, but the measurement trying to be inserted doesn't seem to be hitting it? Note how it doesn't include `date_utc` or `data` but does include `attribution`.

Reply:
I'll check that... this is the first column so it may be that the whole table is somehow badly created... Maybe my knexfile-local.json causes that? Or db is not right? I'll get back to you on this.

Reply:
Fixed.